code/__HELPERS/unsorted.dm

```diff
@@ -1214,3 +1214,24 @@ proc/shift_proximity(atom/checker, atom/A, range, atom/B, newrange)
 		var/turf/F = E
 		F.proximity_checkers |= checker
 	return 1
+
+/proc/flick_overlay_better(image/I, atom/A, duration) //todo :^) change flick_overlay() to this
+	set waitfor = 0
+	if(!A || !I)
+		return
+	A.overlays |= I
+	sleep(duration)
```

Comment:
You do know why this was changed to ```spawn()``` in ```flick_overlay()``` don't you? So you can edit the Image during the flick, so you can, you know, ANIMATE IT.

This is key functionality to attack animations so if you're going to change ```flick_overlay()``` to this (Which I actually disagree with) then you'll need to swap this ```sleep()``` for a ```spawn()```.